### PR TITLE
`rgh-dim-commits` - Add support for new commit view

### LINKED ISSUE #7179: `rgh-dim-commits` don't work on new commits view
### Description

Feature doesn't work with new commits ui.

image

### How to replicate the issue + URL

https://github.com/refined-github/refined-github/commits/main/

### Extension version

23.12.17

### Browser(s) used

Google Chrome & Firefox

### PR DESCRIPTION
**Pull request by Katsute**
Fix #7179

## Test URLs

https://github.com/refined-github/refined-github/commits/98fb24e53704b436d68f254429885fb7bef09541/source/features/bypass-checks.tsx

## Screenshot

image

### CODE REVIEW

File: source/features/rgh-dim-commits.tsx [hunk 1]
```diff
@@ -3,25 +3,23 @@ import * as pageDetect from 'github-url-detection';
...
 function dim(commitTitle: HTMLElement): void {
 	if (excludePreset.test(commitTitle.textContent.trim())) {
-		commitTitle.closest('.js-commits-list-item')!.style.opacity = '50%';
+		commitTitle.closest('[data-testid="commit-row-item"]')!.style.opacity = '50%';
 	}
 }
 
 function init(signal: AbortSignal): void {
-	observe('.js-commits-list-item p:has(> .js-navigation-open)', dim, {signal});
+	observe('[data-testid="commit-row-item"] [data-testid="listview-item-title-container"] span', dim, {signal});
```
Comment (fregante) [inline source/features/rgh-dim-commits.tsx:17]: We can preserve GHE compatibility by keeping both of these selectors
```suggestion
	observe([
		'blah blah', // TODO: Remove in June 2024
		'[data-testid="commit-row-item"] [data-testid="listview-item-title-container"] span'
	], dim, {signal});
```

Also in `[data-testid="commit-row-item"] [data-testid="listview-item-title-container"]` only the latter is necessary, probably.

That final `span` however seems a bit broad. Maybe we can use `span:first-child`?

Comment (Katsute) [reply source/features/rgh-dim-commits.tsx]: Restored original behavior:

image

Comment (Katsute) [reply source/features/rgh-dim-commits.tsx]: I think `[data-testid="listview-item-title-container"] .markdown-title span` is sufficient for the span selector

Comment (fregante) [reply source/features/rgh-dim-commits.tsx]: Note that I suggest to use an array, not to duplicate the observer and handler. We should only do that when the handler differs substantially. In the handler `.closest()` can also take an array of selectors

Comment (Katsute) [reply source/features/rgh-dim-commits.tsx]: neat. didn't know you could do that

Comment (fregante) [reply source/features/rgh-dim-commits.tsx]: Technically it just accepts string, but this is JavaScript, sooooooo… I altered the types in global.d.ts 😃

**Pull request merged**